### PR TITLE
[SAGE-908] Modal add fullscreen variant

### LINKED ISSUE #908: [Improvement] Modal full screen variant
## Description

This cycle includes designs that contain a full screen modal
Rails and React variant needed

#### Design 1
Full screen modal with Title and close button in header
Screen Shot 2021-10-19 at 5 10 30 PM
#### Design 2
Full screen modal with Title, close button, and action button
Screen Shot 2021-10-19 at 5 11 20 PM

### PR DESCRIPTION
**Pull request by QuintonJason**
## Description

- [x] add fullscreen variant to sage modal

## Screenshots

|  No overflow   | Overflow  |
|--------|--------|
|Screen Shot 2021-10-21 at 2 14 00 PM|fullscreenmodal|

## Testing in `sage-lib`

Rails
Visit http://localhost:4000/pages/sandbox and click on the button to see the modal

React
Visit modal storybook
In the options set `fullScreen` to `true`

## Testing in `kajabi-products`

1. (**LOW**) Adding a new property. No affect on the app.

## Related

Closes #908

### CODE REVIEW

**Comment by QuintonJason** [comment]
Will revert sandbox and react storybook before merge

File: packages/sage-assets/lib/stylesheets/components/_modal.scss [hunk 1]
```diff
@@ -10,6 +10,8 @@ $-modal-padding-y: sage-spacing(md);
 $-modal-margin-lg: 6vh;
 $-modal-margin-xl: 8vh;
 $-modal-header-image-size: rem(28px);
+$-modal-fullscreen-padding-y: rem(18px);
```
Comment (philschanely) [inline packages/sage-assets/lib/stylesheets/components/_modal.scss:13]: Do we know for sure `18` is needed here as opposed to one of our existing spacers?

**Review by philschanely — APPROVED** [review]
Nice work here! 🙌

File: packages/sage-assets/lib/stylesheets/components/_modal.scss (hunk 1 above)
Comment (QuintonJason) [reply packages/sage-assets/lib/stylesheets/components/_modal.scss:13]: got the green light for 16, pushing now

**Pull request merged**